src/librbd/cache/ReplicatedWriteLog.h

```diff
@@ -197,21 +208,31 @@ class ReplicatedWriteLog : public ImageCache<ImageCtxT> {
   bool m_barrier_in_progress = false;
   BlockGuardCell *m_barrier_cell = nullptr;
 
+  bool m_wake_up_requested = false;
+  bool m_wake_up_scheduled = false;
+  bool m_wake_up_enabled = true;
   bool m_appending = false;
   bool m_dispatching_deferred_ops = false;
 
+  Contexts m_flush_complete_contexts;
+
   rwl::GenericLogOperations m_ops_to_flush; /* Write ops needing flush in local log */
   rwl::GenericLogOperations m_ops_to_append; /* Write ops needing event append in local log */
 
   /* New entries are at the back. Oldest at the front */
   rwl::GenericLogEntries m_log_entries;
-  rwl::GenericLogEntries m_dirty_log_entries;
+  // rwl::GenericLogEntries m_dirty_log_entries;
+  rwl::GenericWriteLogEntries m_dirty_log_entries;
```

Comment:
Can you provide a quick explanation for the need to switch this to `GenericWriteLogEntries`? I couldn't immediately see any `WriteLogEntry` unique method you were invoking. If this can be kept generic, you can then remove the `get_write_log_entry` helper method (to again avoid a lot of these type casting behavior). If it's just about differentiating writes from flushes, it seems that could be handled by a quick `bool` virtual helper method.

Reply:
@dillaman In your last comments, you said "Can we avoid this cast? Can m_dirty_log_entries be a collection of GenericWriteLogEntry?". That why I switch this to `GenericWriteLogEntries`. Now I switch back to `GenericLogEntries ` and removed `get_write_log_entry `.